pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,469 @@
+		if err != nil {
+			// When the socket is closed in the Run method, this error will be logged, which is not ideal.
+			// In the future, we may try setting a ReadDeadline on the socket before each ReadFrom and using
+			// a channel to signal that the loop should terminate.
+			klog.ErrorS(err, "Failed to read ICMP message")
+			return
+		}
+
+		destIP := peer.String()
+
+		// Parse the ICMP message
+		var msg *icmp.Message
+		if isIPv4 {
+			msg, err = icmp.ParseMessage(protocolICMP, readBuffer[:n])
+			if err != nil {
+				klog.ErrorS(err, "Failed to parse ICMP message")
+				continue
+			}
+			if msg.Type != ipv4.ICMPTypeEcho && msg.Type != ipv4.ICMPTypeEchoReply {
+				klog.V(5).InfoS("Ignoring non-ping ICMP message", "msg", msg)
+				continue
+			}
+			// Ignore ICMP echo messages received from other Nodes (they will be answered by the system)
+			if msg.Type == ipv4.ICMPTypeEcho {
+				klog.V(7).InfoS("Ignoring ICMP echo request message", "msg", msg)
+				continue
+			}
+		} else {
+			msg, err = icmp.ParseMessage(protocolICMPv6, readBuffer)
+			if err != nil {
+				klog.ErrorS(err, "Failed to parse ICMP message")
+				continue
+			}
+			if msg.Type != ipv6.ICMPTypeEchoRequest && msg.Type != ipv6.ICMPTypeEchoReply {
+				klog.V(5).InfoS("Ignoring non-ping ICMP message", "msg", msg)
+				continue
+			}
+			// Ignore ICMP echo messages received from other Nodes (they will be answered by the system)
+			if msg.Type == ipv6.ICMPTypeEchoRequest {
+				klog.V(7).InfoS("Ignoring ICMP echo request message", "msg", msg)
+				continue
+			}
+		}
+
+		echo, ok := msg.Body.(*icmp.Echo)
+		if !ok {
+			klog.ErrorS(nil, "Failed to assert type as *icmp.Echo")
+			continue
+		}
+		if echo.ID != int(icmpEchoID) {
+			klog.V(4).InfoS("Ignoring ICMP message with wrong echo ID", "msg", msg)
+			continue
+		}
+
+		klog.V(4).InfoS("Received ICMP message", "IP", destIP, "msg", msg)
+
+		// Parse the time from the ICMP data
+		sentTime, err := time.Parse(time.RFC3339Nano, string(echo.Data))
+		if err != nil {
+			klog.ErrorS(err, "Failed to parse time from ICMP data")
+			continue
+		}
+
+		// Calculate the round-trip time
+		end := time.Now()
+		rtt := end.Sub(sentTime)
+		klog.V(4).InfoS("Updating latency entry for Node IP", "IP", destIP, "lastSendTime", sentTime, "lastRecvTime", end, "RTT", rtt)
+
+		// Update the latency store
+		mutator := func(entry *NodeIPLatencyEntry) {
+			entry.LastRecvTime = end
+			entry.LastMeasuredRTT = rtt
+		}
+		m.latencyStore.SetNodeIPLatencyEntry(destIP, mutator)
+	}
+}
+
+// pingAll sends ICMP messages to all the Nodes.
+func (m *NodeLatencyMonitor) pingAll(ipv4Socket, ipv6Socket net.PacketConn) {
+	klog.V(4).InfoS("Pinging all Nodes")
+	nodeIPs := m.latencyStore.ListNodeIPs()
+	for _, toIP := range nodeIPs {
+		if toIP.To4() != nil && ipv4Socket != nil {
+			if err := m.sendPing(ipv4Socket, toIP); err != nil {
+				klog.ErrorS(err, "Cannot send ICMP message to Node IP because socket is not initialized for IPv4", "IP", toIP)
```

Comment:
```suggestion
				klog.ErrorS(err, "Cannot send ICMP message to Node IP", "IP", toIP)
```
The extra reason is wrong.

Reply:
OK, I have removed it.